Objects/genericaliasobject.c

```diff
@@ -254,15 +285,46 @@ ga_getitem(PyObject *self, PyObject *item)
         if (typevar) {
             Py_ssize_t iparam = tuple_index(alias->parameters, nparams, arg);
             assert(iparam >= 0);
-            if (is_tuple) {
-                arg = PyTuple_GET_ITEM(item, iparam);
+            arg = args[iparam];
+            Py_INCREF(arg);
+        }
+        else {
+            _Py_IDENTIFIER(__parameters__);
+            PyObject *subparams;
+            if (_PyObject_LookupAttrId(arg, &PyId___parameters__, &subparams) < 0) {
+                Py_DECREF(newargs);
+                return NULL;
+            }
+            if (subparams && PyTuple_Check(subparams)) {
```

Comment:
When could `subparams` be zero here? You already check for failure from the lookup. (Same above in make_parameters() I guess.)

Reply:
It could be zero when `arg` does not have the `__parameters__` attribute.

There are three possibility:
* There is the attribute. `_PyObject_LookupAttrId` returns `1` and sets the output pointer to non-NULL.
* There is no the attribute. `_PyObject_LookupAttrId` returns `0` and sets the output pointer to NULL. No exception is set (if AttributeError was raised, it will be silenced).
* Getting the attribute raises an error. `_PyObject_LookupAttrId` returns `-1`, sets the output pointer to NULL and sets an exception.

It looks a little complicated, but allows to write the most simple code and have the best performance (`AttributeError` is not even raised just to be silenced, and you avoid to call `PyErr_Occurred()` in performance critical code, and explicitly check `PyErr_ExceptionMatches(PyExc_AttributeError)`).

`_PyObject_LookupAttrId` was initially added for performance, but later it replaced `PyObject_GetAttr` in many sites, because it allowed easier to write the correct code (when you need to ignore AttributeError but not other exceptions).